dartdoc_test/lib/src/resource.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:analyzer/file_system/file_system.dart';
+import 'package:analyzer/file_system/memory_file_system.dart';
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:path/path.dart';
+
+class CompositeResourceProvider implements ResourceProvider {
```

Comment:
I'd consider just using the OverlayResourceProvider on top of PhysicalResourceProvider.

Because if you get a `Folder` object from the memoryProvider, then calling [getChild](https://pub.dev/documentation/analyzer/5.13.0/file_system_file_system/Folder/getChild.html) might not return the correct result.

Correctly wrapping a `ResourceProvider` is not easy, if you want to try you can dive into the implementation for `OverlayResourceProvider`. But just using `OverlayResourceProvider` is probably fine. We don't need to inject many files anyways :D

Reply:
I replace it to just use [`OverlayResourcceProvider`](https://pub.dev/documentation/analyzer/5.13.0/file_system_overlay_file_system/OverlayResourceProvider-class.html). I believe a single `resourceProvider` instance should be shared within `dartdoc_test`, but I'm not sure if `analysisContext` or `session` should be share as well... I think I need to understand the `package:analyzer` and anlysis process.